src/lib.rs

```diff
@@ -296,104 +313,93 @@ impl RandomXVM {
         cache: Option<&RandomXCache>,
```

Comment:
nit: I like to move objects when they will be cloned in the method
```suggestion
        cache: Option<RandomXCache>,
        dataset: Option<RandomXDataset>
```